### PR TITLE
Add (better) Sorting

### PR DESCRIPTION
**Pull request by C0D3-M4513R**
Adds Sorting.
Might help with fixing the test from #2948 

There are a lot of issues, where the lint thinks lines are duplicated, but I do not see a better way.

### CODE REVIEW

**Review by spiffcs — COMMENTED** [review]
I've gotten through about 40/80 files changed here and made a few comments for myself to come back to along with a question about keeping the current sorting ordering behavior so we don't affect current users. I'll come back when I have some cycles to finish this review.

File: syft/format/cyclonedxxml/test-fixtures/snapshot/TestCycloneDxImageEncoder.golden [hunk 1]
```diff
@@ -17,6 +17,20 @@
     </component>
   </metadata>
   <components>
+    <component bom-ref="redacted" type="library">
+      <name>package-2</name>
```
[the remark this answers is not in the document]
Comment (C0D3-M4513R) [reply syft/format/cyclonedxxml/test-fixtures/snapshot/TestCycloneDxImageEncoder.golden:21]: I can definitely look into the feasibility of keeping the ordering the same.
I don't expect it to be that difficult.

[the remark this answers is not in the document]
Comment (C0D3-M4513R) [reply syft/format/cyclonedxxml/test-fixtures/snapshot/TestCycloneDxImageEncoder.golden]: I've been trying to make the sort stable, but I can't quite make all test-cases pass.
There are always some outliers. I have done the best I could I think. 

Just out of curiosity: which part about backwards compatibility is being broken, if package and relations were sorted differently? Are you considering the output order as relevant to semantic versioning?

File: syft/format/text/test-fixtures/snapshot/TestTextImageEncoder.golden [hunk 2]
```diff
@@ -9,13 +9,13 @@
  Size:		 16
  MediaType:	 application/vnd.docker.image.rootfs.diff.tar.gzip
 
-[package-1]
- Version:	 1.0.1
- Type:		 python
- Found by:	 the-cataloger-1
-
 [package-2]
```
[the remark this answers is not in the document]
Comment (C0D3-M4513R) [reply syft/format/text/test-fixtures/snapshot/TestTextImageEncoder.golden]: This specific instance has been fixed.
The remaining tests, where orderings have changed are:
- syft/format/common/cyclonedxhelpers/to_format_model_test.go
- syft/format/common/spdxhelpers/to_syft_model_test.go
- syft/format/spdxjson/test-fixtures/snapshot/TestSPDX22JSONRequredProperties.golden
- syft/format/spdxtagvalue/test-fixtures/snapshot/TestSPDXJSONSPDXIDs.golden

**Comment by C0D3-M4513R** [comment]
Todo: to be replaced with a better version, more tailored towards relationship sorting in tests with less code.

**Pull request closed without merging**